cobra/core/DictList.py

```diff
@@ -54,7 +54,7 @@ def list_attr(self, attribute):
         """
         return [getattr(i, attribute) for i in self]
 
-    def query(self, search_function, attribute="id"):
+    def query(self, search_function, attribute=None):
```

Comment:
yes exactly, that's what I meant, to cover the situation when you only give a string there should be a sensible behaviour. If you give a function it makes sense to have to set the attribute.